### PR TITLE
added 10Q info table

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: insights-ui/src/app/api/tickers/route.ts [hunk 1]
```diff
@@ -38,6 +38,28 @@ async function postHandler(req: NextRequest): Promise<Ticker> {
     data,
   });
 
+  const url = 'https://4mbhgkl77s4gubn7i2rdcllbru0wzyxl.lambda-url.us-east-1.on.aws/reporting_period_and_filing_link';
+  const payload = { ticker: tickerKey };
+  const ReportingPeriodAndFilingLinkResponse = await fetch(url, {
```
Comment (RobinNagpal) [inline insights-ui/src/app/api/tickers/route.ts:43]: In typescript the variables don't start with a capital letters. These are basic things which should not be pointed out

File: insights-ui/src/app/api/tickers/route.ts [hunk 2]
```diff
@@ -38,6 +38,28 @@ async function postHandler(req: NextRequest): Promise<Ticker> {
...
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(payload),
+  });
+  const ReportingPeriodAndFilingLink = await ReportingPeriodAndFilingLinkResponse.json();
+  console.log('ReportingPeriodAndFilingLink', ReportingPeriodAndFilingLink);
```
Comment (RobinNagpal) [inline insights-ui/src/app/api/tickers/route.ts:49]: Remove console log unless really needed. These are basic things which should not be pointed out

File: insights-ui/src/app/api/tickers/route.ts [hunk 3]
```diff
@@ -38,6 +38,28 @@ async function postHandler(req: NextRequest): Promise<Ticker> {
...
+  if ('message' in ReportingPeriodAndFilingLink) {
+    throw new Error(ReportingPeriodAndFilingLink.message);
```
Comment (RobinNagpal) [inline insights-ui/src/app/api/tickers/route.ts:51]: Why do we need this? Will the `ReportingPeriodAndFilingLinkResponse.json();` not throw an error? if the response is not successful?

File: insights-ui/src/app/api/actions/tickers/[tickerKey]/ten-q-info/route.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { prisma } from '@/prisma';
+import { withErrorHandlingV2 } from '@dodao/web-core/api/helpers/middlewares/withErrorHandling';
+import { Latest10QInfo } from '@prisma/client';
+import { NextRequest } from 'next/server';
+
+async function getLatest10QInfoHandler(req: NextRequest, { params }: { params: Promise<{ tickerKey: string }> }): Promise<Latest10QInfo> {
```
Comment (RobinNagpal) [inline insights-ui/src/app/api/actions/tickers/[tickerKey]/ten-q-info/route.ts:6]: We dont need an extra route for this. We can get it as part of ticker information. See the ticker details route. we have many `include` items there

File: insights-ui/src/app/api/tickers/route.ts [hunk 5]
```diff
@@ -38,6 +38,28 @@ async function postHandler(req: NextRequest): Promise<Ticker> {
...
+  }
+
+  const newLatest10QInfo = await prisma.latest10QInfo.create({
```
Comment (RobinNagpal) [inline insights-ui/src/app/api/tickers/route.ts:54]: How do we plan to populate the information for existing tickers?

I think we discussed it in the chat? Then why its not there in the code?

image
